lib/spack/spack/test/versions.py

```diff
@@ -614,3 +614,17 @@ def test_empty_version_range_raises():
         assert VersionRange('2', '1.0')
     with pytest.raises(ValueError):
         assert ver('2:1.0')
+
+
+@pytest.mark.regression('26482')
+def test_version_list_with_range_included_in_concrete_version_interpreted_as_range():
+    # Note: this test only tests whether we can construct a version list of a range
+    # and a version, where the range is contained in the version when it is interpreted
+    # as a range. That is: Version('3.1') interpreted as VersionRange('3.1', '3.1').
+    # Cleary it *shouldn't* be interpreted that way, but that is how Spack currently
+    # behaves, and this test only ensures that creating a VersionList of this type
+    # does not throw like reported in the linked Github issue.
+    v = VersionList([Version('3.1'), VersionRange('3.1.1', '3.1.2')])
+
+    # This currently does not hold, but should hold.
+    # assert not v.concrete
```

Comment:
Can you uncomment the assert statement and mark the test as xfail?

Reply:
I've split it in two tests so that we at least know that VersionList can be constructed even if incorrect